File: frontend/src/types/product.ts
export interface PriceData {
  price: number;
  original_price: number | null;
  discount_pct: number | null;
  in_stock: boolean;
  scraped_at: string;
}

export interface ProductResponse {
  id: string;
  url: string;
  platform: "trendyol" | "hepsiburada" | string;
  name: string;
  brand: string | null;
  category: string | null;
  image_url: string | null;
  created_at: string;
  latest_price: PriceData;
}

export interface ProductListResponse {
  products: ProductResponse[];
  total: number;
}

export interface PriceHistoryEntry {
  price: number;
  original_price: number | null;
  discount_pct: number | null;
  in_stock: boolean;
  scraped_at: string;
}

export interface PriceStatsResponse {
  product_id: string;
  days: number;
  min_price: number | null;
  max_price: number | null;
  avg_price: number | null;
  stddev_price: number | null;
  data_points: number;
}

export interface ForecastPoint {
  date: string;
  predicted_price: number;
}

export interface ForecastResponse {
  product_id: string;
  forecast_days: number;
  forecast: ForecastPoint[];
  current_price: number;
  predicted_final_price: number;
  mae: number;
  low_confidence: boolean;
  recommendation: "AL" | "BEKLE" | "TAKIPTE KAL";
  data_points: number;
}

File: backend/app/schemas/product.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, HttpUrl


class ProductSubmitRequest(BaseModel):
    url: HttpUrl


class PriceDataResponse(BaseModel):
    price: float
    original_price: float | None
    discount_pct: float | None
    in_stock: bool
    scraped_at: datetime


class ProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    url: str
    platform: str
    name: str
    brand: str | None
    category: str | None
    image_url: str | None
    created_at: datetime
    latest_price: PriceDataResponse


class ProductListResponse(BaseModel):
    products: list[ProductResponse]
    total: int


class PriceHistoryEntry(BaseModel):
    price: float
    original_price: float | None
    discount_pct: float | None
    in_stock: bool
    scraped_at: datetime


class PriceStatsResponse(BaseModel):
    product_id: uuid.UUID
    days: int
    min_price: float | None
    max_price: float | None
    avg_price: float | None
    stddev_price: float | None
    data_points: int
